ocaml/xcp-rrdd/bin/rrdp-netdev/rrdp_netdev.ml

```diff
@@ -18,6 +18,128 @@ module D = Debug.Make (struct let name = "xcp-rrdp-netdev" end)
 
 module Process = Rrdd_plugin.Process (struct let name = "xcp-rrdd-netdev" end)
 
+type iface_stats = {
+    tx_bytes: int64  (** bytes emitted *)
+  ; tx_pkts: int64  (** packets emitted *)
+  ; tx_errors: int64  (** error emitted *)
+  ; rx_bytes: int64  (** bytes received *)
+  ; rx_pkts: int64  (** packets received *)
+  ; rx_errors: int64  (** error received *)
+}
+
+let default_stats =
+  {
+    tx_bytes= 0L
+  ; tx_pkts= 0L
+  ; tx_errors= 0L
+  ; rx_bytes= 0L
+  ; rx_pkts= 0L
+  ; rx_errors= 0L
+  }
+
+let monitor_whitelist =
+  ref
+    [
+      "eth"
+    ; "vif" (* This includes "tap" owing to the use of standardise_name below *)
+    ]
+
+let standardise_name name =
+  try
+    let d1, d2 = Scanf.sscanf name "tap%d.%d" (fun d1 d2 -> (d1, d2)) in
+    let newname = Printf.sprintf "vif%d.%d" d1 d2 in
+    newname
+  with _ -> name
```

Comment:
Do we want to log this? It means sscanf failed to scan name

Reply:
it will fail to scan for everything but "tapX.X", this is intentional to only transform those names

Reply:
Maybe add a comment because it is not obvious